File: frontend/src/types/product.ts
export interface Product {
  id: string;
  name: string;
  description?: string;
  tagline?: string;
  logo_url?: string;
  website?: string;
  created_at: string;
  updated_at: string;
  segments: ProductSegment[];
  modules: ProductModule[];
}

export interface ProductCreate {
  name: string;
  description?: string;
  tagline?: string;
  logo_url?: string;
  website?: string;
}

export interface ProductUpdate {
  name?: string;
  description?: string;
  tagline?: string;
  logo_url?: string;
  website?: string;
}

export interface ProductSegment {
  id: string;
  product_id: string;
  name: string;
  description?: string;
  target_market?: string;
  customer_size?: CustomerSize;
  created_at: string;
  updated_at: string;
}

export interface ProductModule {
  id: string;
  product_id: string;
  name: string;
  description?: string;
  icon?: string;
  order_index: number;
  created_at: string;
  updated_at: string;
  feature_count?: number;
}

export enum CustomerSize {
  SMB = 'SMB',
  MID_MARKET = 'Mid Market',
  ENTERPRISE = 'Enterprise',
  ALL = 'All',
}

export interface BattleCard {
  id: string;
  product_id: string;
  competitor_id: string;
  version: number;
  status: BattleCardStatus;
  published_at?: string;
  created_by?: string;
  created_at: string;
  updated_at: string;
  sections: BattleCardSection[];
  product_name?: string;
  competitor_name?: string;
}

export interface BattleCardSection {
  id: string;
  battle_card_id: string;
  section_type: BattleCardSectionType;
  content: any;
  order_index: number;
  created_at: string;
  updated_at: string;
}

export enum BattleCardStatus {
  DRAFT = 'Draft',
  PUBLISHED = 'Published',
  ARCHIVED = 'Archived',
}

export enum BattleCardSectionType {
  WHY_WE_WIN = 'Why We Win',
  COMPETITOR_STRENGTHS = 'Competitor Strengths',
  OBJECTION_HANDLING = 'Objection Handling',
  FEATURE_COMPARISON = 'Feature Comparison',
  PRICING_COMPARISON = 'Pricing Comparison',
  KEY_DIFFERENTIATORS = 'Key Differentiators',
}

export interface ScrapingJob {
  id: string;
  competitor_id: string;
  job_type: ScrapingJobType;
  status: ScrapingJobStatus;
  started_at?: string;
  completed_at?: string;
  error_message?: string;
  results?: any;
  created_at: string;
  updated_at: string;
}

export enum ScrapingJobType {
  FEATURES = 'Features',
  PRICING = 'Pricing',
  NEWS = 'News',
  REVIEWS = 'Reviews',
  FULL_SCAN = 'Full Scan',
}

export enum ScrapingJobStatus {
  PENDING = 'Pending',
  RUNNING = 'Running',
  COMPLETED = 'Completed',
  FAILED = 'Failed',
}

File: backend/app/schemas/product.py
from typing import List, Optional, Dict, Any
from datetime import datetime
from uuid import UUID
from pydantic import BaseModel

from app.models.enums import CustomerSize, BattleCardStatus, BattleCardSectionType, ScrapingJobType, ScrapingJobStatus


# Product Schemas
class ProductCreate(BaseModel):
    name: str
    description: Optional[str] = None
    tagline: Optional[str] = None
    logo_url: Optional[str] = None
    website: Optional[str] = None


class ProductUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    tagline: Optional[str] = None
    logo_url: Optional[str] = None
    website: Optional[str] = None


class ProductResponse(BaseModel):
    id: UUID
    name: str
    description: Optional[str]
    tagline: Optional[str]
    logo_url: Optional[str]
    website: Optional[str]
    created_at: datetime
    updated_at: datetime
    segments: List["ProductSegmentResponse"] = []
    modules: List["ProductModuleResponse"] = []

    class Config:
        from_attributes = True


# Product Segment Schemas
class ProductSegmentCreate(BaseModel):
    name: str
    description: Optional[str] = None
    target_market: Optional[str] = None
    customer_size: Optional[CustomerSize] = None


class ProductSegmentUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    target_market: Optional[str] = None
    customer_size: Optional[CustomerSize] = None


class ProductSegmentResponse(BaseModel):
    id: UUID
    product_id: UUID
    name: str
    description: Optional[str]
    target_market: Optional[str]
    customer_size: Optional[CustomerSize]
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


# Product Module Schemas
class ProductModuleCreate(BaseModel):
    name: str
    description: Optional[str] = None
    icon: Optional[str] = None
    order_index: int = 0


class ProductModuleUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    icon: Optional[str] = None
    order_index: Optional[int] = None


class ProductModuleResponse(BaseModel):
    id: UUID
    product_id: UUID
    name: str
    description: Optional[str]
    icon: Optional[str]
    order_index: int
    created_at: datetime
    updated_at: datetime
    feature_count: int = 0

    class Config:
        from_attributes = True


class ModuleReorderRequest(BaseModel):
    module_orders: List[Dict[str, Any]]  # [{"id": "uuid", "order_index": 0}, ...]


# Battle Card Schemas
class BattleCardSectionCreate(BaseModel):
    section_type: BattleCardSectionType
    content: Dict[str, Any]
    order_index: int = 0


class BattleCardSectionUpdate(BaseModel):
    section_type: Optional[BattleCardSectionType] = None
    content: Optional[Dict[str, Any]] = None
    order_index: Optional[int] = None


class BattleCardSectionResponse(BaseModel):
    id: UUID
    battle_card_id: UUID
    section_type: BattleCardSectionType
    content: Dict[str, Any]
    order_index: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class BattleCardCreate(BaseModel):
    product_id: UUID
    competitor_id: UUID
    sections: List[BattleCardSectionCreate] = []


class BattleCardUpdate(BaseModel):
    status: Optional[BattleCardStatus] = None
    sections: Optional[List[BattleCardSectionUpdate]] = None


class BattleCardResponse(BaseModel):
    id: UUID
    product_id: UUID
    competitor_id: UUID
    version: int
    status: BattleCardStatus
    published_at: Optional[datetime]
    created_by: Optional[UUID]
    created_at: datetime
    updated_at: datetime
    sections: List[BattleCardSectionResponse] = []
    product_name: Optional[str] = None
    competitor_name: Optional[str] = None

    class Config:
        from_attributes = True


class BattleCardGenerateRequest(BaseModel):
    competitor_id: UUID
    include_sections: List[BattleCardSectionType] = [
        BattleCardSectionType.WHY_WE_WIN,
        BattleCardSectionType.COMPETITOR_STRENGTHS,
        BattleCardSectionType.OBJECTION_HANDLING,
        BattleCardSectionType.FEATURE_COMPARISON,
        BattleCardSectionType.KEY_DIFFERENTIATORS
    ]


# Scraping Job Schemas
class CompetitorScrapingRequest(BaseModel):
    job_type: ScrapingJobType
    target_urls: Optional[List[str]] = []


class ScrapingJobResponse(BaseModel):
    id: UUID
    competitor_id: UUID
    job_type: ScrapingJobType
    status: ScrapingJobStatus
    started_at: Optional[datetime]
    completed_at: Optional[datetime]
    error_message: Optional[str]
    results: Optional[Dict[str, Any]]
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


# Feature Assignment
class FeatureModuleAssignment(BaseModel):
    feature_id: UUID
    module_id: Optional[UUID]
    is_key_differentiator: bool = False


class ModuleFeaturesUpdate(BaseModel):
    feature_assignments: List[FeatureModuleAssignment]


# Update forward references
ProductResponse.model_rebuild()
BattleCardResponse.model_rebuild()